### PR TITLE
Fix regex that causes usernames with `-` chars to skip meta replication

### PR DESCRIPTION
**Pull request by mrsarm**
# Description

Ticket: https://github.com/medic/cht-core/issues/6591

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: sentinel/src/schedule/replications.js [hunk 1]
```diff
@@ -121,7 +121,7 @@ module.exports = {
     return Promise.resolve();
   },
   runReplication: () => {
-    const SRC_DB_REGEX = new RegExp(`${db.medicDbName}-user-[^\\-]+-meta`);
+    const SRC_DB_REGEX = new RegExp(`${db.medicDbName}-user-[a-z0-9_-]+-meta`);
```
Comment (garethbowen) [inline sentinel/src/schedule/replications.js:124]: I think this is going to miss other potentially valid user DBs. In API we have [a regex](https://github.com/medic/cht-core/blob/master/api/src/services/user-db.js#L12) to detect and escape characters in user names that are not valid in DB names so we have to accept at least all those characters.

For example, this perfectly valid DB name fails this regex: `medic-user-stupid+name-meta`.

In fact I think we can be very lenient here - as long as the DB name starts with `${db.medicDbName}-user-` and ends with `-meta` we should be good, so what about just...

```suggestion
    const SRC_DB_REGEX = new RegExp(`${db.medicDbName}-user-.+-meta`);
```

**Review by garethbowen — CHANGES_REQUESTED** [review]
Nice! I think we can be even more lenient about the db name.

Also, we should definitely add unit tests to check a bunch of db names to ensure the regex is working as expected. If it's a pain to add unit tests for the whole function, consider pulling out a new function like `isUserMetaDb` which just runs this one regex so testing is simple.

File: sentinel/src/schedule/replications.js (hunk 1 above)
Comment (mrsarm) [reply sentinel/src/schedule/replications.js:124]: Actually this regex `[a-z0-9_-]` is the same used to validate users when created: https://github.com/medic/cht-core/blob/df35966a8a815063e294869b3d6858ee413498a3/api/src/services/users.js#L18

If you try to create a user like the one in your example you will get an error like:

cht-error-username

Are you sure you want to open the regex to forbidden characters?

Comment (garethbowen) [reply sentinel/src/schedule/replications.js:124]: Yes, make the one in the replication code more tolerant. It's possible users were created either before the check was enforced in the UI, or directly via CURL or medic-conf or fauxton.

**Review by garethbowen — CHANGES_REQUESTED** [review]
Awesome. Code looks great.

Now it just needs a few unit tests to ensure it works and we don't regress later.

**Comment by mrsarm** [comment]
> Now it just needs a few unit tests to ensure it works and we don't regress later.

@garethbowen is this a retaliation because what I said about Sinon ? :stuck_out_tongue:

**Comment by mrsarm** [comment]
Instead of add a new test, I have modified one that already checks whether the DB filtering was working OK.

**Review by garethbowen — APPROVED** [review]
Awesome!

**Comment by garethbowen** [comment]
> is this a retaliation because what I said about Sinon ?

It'll grow on you... like a fungus.

**Pull request merged**